File: go/p0007/p0007_test.go
package p0007

import (
	"testing"

	"github.com/stretchr/testify/suite"
)

type result struct {
	arg1   int
	target int
}

var values = []result{
	{
		arg1:   123,
		target: 321,
	},
	{
		arg1:   -123,
		target: -321,
	},
	{
		arg1:   120,
		target: 21,
	},
}

type p0007TestSuite struct {
	suite.Suite
}

func (s *p0007TestSuite) Test() {
	for _, v := range values {
		s.Equal(v.target, reverse(v.arg1))
	}
}

func TestP0007TestSuite(t *testing.T) {
	s := &p0007TestSuite{}
	suite.Run(t, s)
}

File: js/p0007/p0007.test.js
import reverse from './p0007';

test('reverse a number', () => {
  const argLen = reverse.length;
  const testArr = [
    {
      arg1: 123,
      target: 321,
    },
    {
      arg1: -123,
      target: -321,
    },
    {
      arg1: 2147483647,
      target: 0,
    },
  ];
  testArr.forEach((el) => {
    const args = [];
    for (let i = 1; i <= argLen; i += 1) {
      const keyStr = `arg${i}`;
      args.push(el[keyStr]);
    }

    expect(reverse.apply(this, args)).toEqual(el.target);
  });
});
